runtime/src/lib.rs

```diff
@@ -272,12 +272,18 @@ impl pallet_node_authorization::Config for Runtime {
     type WeightInfo = ();
 }
 
+parameter_types! {
+    pub const MaxMetadataCount: u32 = 16;
+}
+
 /// Configure the template pallet in pallets/simple-nft.
 impl pallet_simple_nft::Config for Runtime {
     type Event = Event;
     type TokenId = u128;
-    type TokenMetadata = Hash;
+    type TokenMetadataKey = Vec<u8>;
```

Comment:
I wonder if this would be better as a length limited array, `[u8; 32]` say

Reply:
changed, still `u64` for tests

Reply:
and updated API to handle padding